plugin/src/main/kotlin/com/automattic/android/applystyle/ApplyStylePlugin.kt

```diff
@@ -0,0 +1,167 @@
+/*
+ * This Kotlin source file was generated by the Gradle 'init' task.
+ */
+package com.automattic.android.applystyle
+
+import org.gradle.api.Project
+import org.gradle.api.Plugin
+
+import io.gitlab.arturbosch.detekt.DetektPlugin
+import io.gitlab.arturbosch.detekt.extensions.DetektExtension
+import org.gradle.api.plugins.quality.CodeQualityExtension
+import java.io.File
+import java.io.IOException
+import java.nio.file.Path
+
+/**
+ * A custom plugin that standardizes the styles across various projects.
+ *
+ * For now the plugin deals only with the configuration of Detekt.
+ */
+class ApplyStylePlugin: Plugin<Project> {
+    override fun apply(project: Project) {
+        project.tasks.register("extractDetektConf") { task ->
+            task.doLast {
+                val detektExtension = project.getDetektExtension() as DetektExtension
+                val file = detektExtension.config.singleFile ?: path(project.buildDir,"detekt.yml")
+                project.extractDetektConf(file, true)
+                println("Wrote detekt configuration file to $file")
+            }
+        }
+
+        project.tasks.register("greeting") { task ->
+            task.doLast {
+                println("Hello from plugin '$PLUGIN_NAME'")
+            }
+        }
+
+        project.plugins.withId("kotlin-android") {
+            project.applyDetektPlugin()
+        }
+    }
+
+    /**
+     * Applies the detekt plugin to the project and configures the project properly.
+     */
+    private fun Project.applyDetektPlugin() {
+        debug("==> applying detekt plugin")
+        project.pluginManager.apply(DetektPlugin::class.java)
+
+        project.applyDetektDefaults()
+        project.afterEvaluate {
+            project.writeDetektConf()
+        }
+    }
+
+    /**
+     * Applies our detekt default settings.
+     */
+    private fun Project.applyDetektDefaults() {
+        val detektExtension = project.getDetektExtension() as DetektExtension
+        with (detektExtension) {
+            autoCorrect = false
+            buildUponDefaultConfig = false
```

Comment:
Question (❓): Unless I am mistaken the `buildUponDefaultConfig` needs to be set to `true` right? Otherwise, running against the for example test [potyl:a8c-apply-style-gradle-plugin](https://github.com/potyl/WordPress-Android/tree/a8c-apply-style-gradle-plugin) branch will give a false positive that Detekt is successful, right?

PS: I actually tried that myself and it seems that this is indeed the case. After enabling `buildUponDefaultConfig` the [potyl:a8c-apply-style-gradle-plugin](https://github.com/potyl/WordPress-Android/tree/a8c-apply-style-gradle-plugin) Detekt task failed with a few warning, which is what I would expect. After me adding a custom `detekt.yml` file with those needed changes to the rules, to override the default ones from `buildUponDefaultConfig`, Detekt started passing again, see below:

`config/detekt.yml`
```
complexity:
  LongParameterList:
    ignoreAnnotated: ['javax.inject.Inject', 'javax.annotation.Generated']

formatting:
  android: true
  autoCorrect: false

style:
  ForbiddenComment:
    values: ['FIXME:', 'STOPSHIP:']
  MagicNumber:
    ignoreEnums: true
```

To be the above exercise means that the plugin works as I expected, that is after it's been configured, per project and as expected, good job! 🌟